rust/otap-dataflow/crates/otlp/src/otlp_batch_processor.rs

```diff
@@ -24,57 +24,51 @@ pub trait HierarchicalBatchSplit: Sized {
 impl HierarchicalBatchSplit for ExportTraceServiceRequest {
     fn split_into_batches(mut self, max_batch_size: usize) -> Vec<Self> {
         let mut batches = Vec::new();
```

Comment:
It would be good to calculate the number of spans beforehand to pre-allocate this vec with just the right capacity. That way could avoid any surprises with `Vec::push`. Something like this:
```rust
let total_spans = self.resource_spans
    .iter()
    .flat_map(|rs| &rs.scope_spans)
    .map(|ss| ss.spans.len())
    .sum::<usize>();
let estimated_batches = (total_spans / max_batch_size) + 1;
let mut batches = Vec::with_capacity(estimated_batches);
```

Reply:
```suggestion
        // TODO: pre-allocate a sized vector
        let mut batches = Vec::new();
```